internal/api/grpc_service.go

```diff
@@ -443,6 +445,75 @@ func (s *GrpcService) GetDeviceDefinitionAll(ctx context.Context, in *emptypb.Em
 	return result, nil
 }
 
+func (s *GrpcService) GetDeviceCompatibilities(ctx context.Context, in *p_grpc.GetDeviceCompatibilityListRequest) (*p_grpc.GetDeviceCompatibilityListResponse, error) {
+	qryResult, _ := s.Mediator.Send(ctx, &queries.GetDeviceCompatibilityQuery{
+		MakeID:        in.MakeId,
+		IntegrationID: in.IntegrationId,
+		Region:        in.Region,
+	})
+
+	deviceCompatibilities := qryResult.(queries.GetDeviceCompatibilityQueryResult)
+
+	result := &p_grpc.GetDeviceCompatibilityListResponse{}
+
+	integFeats := deviceCompatibilities.IntegrationFeatures
+	dcMap := make(map[string][]*p_grpc.DeviceCompatibilities)
+	for _, v := range deviceCompatibilities.DeviceDefinitions {
+		if len(v.R.DeviceIntegrations) == 0 {
+			s.logger.Debug().
+				Str("Model", v.Model).
+				Str("DeviceDefinition", v.ID).
+				Msg("Could not find device integrations")
+			continue
+		}
+
+		di := v.R.DeviceIntegrations[0]
+
+		if di.Features.IsZero() {
+			s.logger.Debug().
+				Str("Model", v.Model).
+				Str("DeviceDefinition", v.ID).
+				Msg("No compatibility information found")
+			continue
+		}
+		res := &p_grpc.DeviceCompatibilities{Year: int32(v.Year)}
+
+		feats := []*p_grpc.Feature{}
+		var features []elasticModels.DeviceIntegrationFeatures
+
+		err := di.Features.Unmarshal(&features)
+		if err != nil {
+			s.logger.Debug().
+				Str("Model", v.Model).
+				Str("DeviceDefinition", v.ID).
+				Msg("Error de-serializing features information")
+			continue
+		}
+
+		for _, f := range features {
+			fkey := f.FeatureKey
+			if fkey == "tires" {
+				fkey = "tires.frontLeft"
```

Comment:
I know we know why this is the case, but good place to have a comment for future ref

Reply:
I'm not sure why this internal endpoint beneath this call is exposing the Elastic column name. Why not use the feature key everywhere where you're not directly hitting Elastic?